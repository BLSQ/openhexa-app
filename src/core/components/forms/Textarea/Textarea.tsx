import clsx from "clsx";
import { TextareaHTMLAttributes } from "react";

type TextareaProps = TextareaHTMLAttributes<HTMLTextAreaElement> & {
  error?: string | null | false | true;
};

const Textarea = (props: TextareaProps) => {
  const { className, error, rows = 4, value, ...delegated } = props;
  return (
    <textarea
      rows={rows}
      className={clsx(
        "form-textarea w-full appearance-none rounded-md border px-3 py-2 focus:z-10 focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 sm:text-sm",
        error
          ? "border-red-300  text-red-900  placeholder-red-300  focus:border-red-500  focus:ring-red-500"
          : "border-gray-300 text-gray-900 placeholder-gray-500 focus:border-blue-500 focus:ring-blue-500",
        className,
      )}
      value={value}
      {...delegated}
    />
  );
};

export default Textarea;
