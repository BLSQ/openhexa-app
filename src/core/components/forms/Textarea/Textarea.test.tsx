import Textarea from "./Textarea";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Textarea", () => {
  it("renders", async () => {
    const { container } = render(
      <Textarea value={"this is a textarea"} onChange={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("calls onChange when user types in the textarea", async () => {
    const onChange = jest.fn();
    const { container } = render(
      <Textarea
        data-testid="textarea"
        value={"this is a textarea"}
        onChange={onChange}
      />
    );
    const textarea = screen.getByTestId("textarea");

    await userEvent.type(textarea, "TEXT");

    expect(onChange).toHaveBeenCalled();
  });
});
