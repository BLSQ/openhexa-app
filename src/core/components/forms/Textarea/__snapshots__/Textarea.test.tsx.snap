// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea renders 1`] = `
<div>
  <textarea
    class="form-textarea w-full appearance-none rounded-md border px-3 py-2 focus:z-10 focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 sm:text-sm border-gray-300 text-gray-900 placeholder-gray-500 focus:border-blue-500 focus:ring-blue-500"
    rows="4"
  >
    this is a textarea
  </textarea>
</div>
`;
